dag_in_context/src/greedy_dag_extractor.rs

```diff
@@ -40,11 +40,11 @@ pub(crate) struct Extractor<'a> {
 
     // Each term must correspond to a node in the egraph. We store that here
     // Use an indexmap for deterministic order of iteration
-    pub(crate) correspondence: IndexMap<Term, NodeId>,
+    pub(crate) correspondence: IndexMap<Term, HashSet<NodeId>>,
```

Comment:
Isn't there a 1-1 correspondence between term and node_id?

Reply:
Nope- since we erase contexts, terms may correspond to multiple nodes